core/resource/resource.go

```diff
@@ -74,6 +86,18 @@ type PagedResources struct {
 
 type ResourcePermissions = map[string][]string
 
+type ResourceConfig struct {
```

Comment:
this will be confusing with the actual config. probably we need to name it differently, something like `Config`? or different naming if we put the config under `config` package here